zookeeper-server/src/main/java/org/apache/zookeeper/server/watch/WatcherCleaner.java

```diff
@@ -133,7 +132,7 @@ public void run() {
                     // same time in the quorum
                     if (!stopped && deadWatchers.size() < watcherCleanThreshold) {
                         int maxWaitMs = (watcherCleanIntervalInSeconds
-                                         + r.nextInt(watcherCleanIntervalInSeconds / 2 + 1)) * 1000;
+                                         + ThreadLocalRandom.current().nextInt(watcherCleanIntervalInSeconds / 2 + 1)) * 1000;
```

Comment:
- I also found `BlueThrottle` `ZooKeeperServer` still have the old `Random`, they also needs to change to `ThreadLocalRandom.current()`?
- Please add a description about why needs this change? e.g. 
`ThreadLocalRandom.current() is thread-safe and have a better performance than the old Random to avoid the Race Condition of CAS by multiple-threads
`

Reply:
Hello @maoling and thank you yet again for the review.

I will address checkstyle violations.

In regards to the places that I left, they take advantage of the ability to set a `seed` for Random for testing purposes.  I didn't want to change that capability.

Reply:
Hi @belugabehr, I wrote BlueThrottleTest and it didn't use seed. It overwrites the nextDouble() function of Random so it returns a non-random number for the test to get repeatable results. We probably can do the same for ThreadLocalRandom so it can be used?